px4_ros2_cpp/src/control/setpoint_types/experimental/attitude.cpp

```diff
@@ -37,6 +37,40 @@ void AttitudeSetpointType::update(
   _vehicle_attitude_setpoint_pub->publish(sp);
 }
 
+void AttitudeSetpointType::update(
+  const float roll,
+  const float pitch,
+  const float yaw,
+  const Eigen::Vector3f & thrust_setpoint_body,
+  const float yaw_sp_move_rate_rad_s)
+{
+  onUpdate();
+
+  px4_msgs::msg::VehicleAttitudeSetpoint sp{};
+  sp.timestamp = _node.get_clock()->now().nanoseconds() / 1000;
+
+  sp.roll_body = roll;
+  sp.pitch_body = pitch;
+  sp.yaw_body = yaw;
+
+  sp.yaw_sp_move_rate = yaw_sp_move_rate_rad_s;
+
+  Eigen::Quaternionf att_setpoint_q{Eigen::AngleAxisf(roll, Eigen::Vector3f::UnitX()) *
```

Comment:
You can use the util methods like here: https://github.com/Auterion/px4-ros2-interface-lib/pull/23/files#diff-73bebd6da924b3ed229ab58e3a1507dd06b157139b1b5afa52a4f41f6d6c6d58R175